core/src/main/java/hudson/slaves/SlaveComputer.java

```diff
@@ -482,7 +481,15 @@ public void onClosed(Channel c, IOException cause) {
                     taskListener.getLogger().println("Connection terminated");
                 }
                 closeChannel();
-                launcher.afterDisconnect(SlaveComputer.this, taskListener);
+                try {
+                    launcher.afterDisconnect(SlaveComputer.this, taskListener);
+                } catch (Throwable t) {
+                    LogRecord lr = new LogRecord(Level.SEVERE,
+                            "Launcher {0}'s afterDisconnect method propagated an exception when {1}'s connection was closed: {2}");
+                    lr.setThrown(t);
+                    lr.setParameters(new Object[]{launcher, SlaveComputer.this, t.getMessage()});
```

Comment:
> then somebody decided to change logging level and it will be slow logging

So under a condition I have never seen in the wild (which would probably require some Jenkins core patches to even implement, since `/log/all` is hardcoded to log everything at `INFO` or above), this `catch` statement which is probably encountered infrequently will take another few microseconds. Hardly justification for longer and harder-to-read code.